lahja/tools/benchmark/process.py

```diff
@@ -92,36 +96,37 @@ def start(self) -> None:
         self._process.start()
 
     @staticmethod
-    async def worker(event_bus: Endpoint, num_events: int) -> None:
-        counter = itertools.count(1)
-        stats = LocalStatistic()
-        async for event in event_bus.stream(PerfMeasureEvent):
-            stats.add(RawMeasureEntry(
-                sent_at=event.sent_at,
-                received_at=time.time()
-            ))
-
-            if next(counter) == num_events:
-                event_bus.broadcast(
-                    TotalRecordedEvent(stats.crunch(event_bus.name)),
-                    BroadcastConfig(filter_endpoint=REPORTER_ENDPOINT)
-                )
-                event_bus.stop()
-                break
+    async def worker(name: str, num_events: int) -> None:
+        with Endpoint() as event_bus:
+            await event_bus.start_serving(ConnectionConfig.from_name(name))
+            await event_bus.connect_to_endpoints(ConnectionConfig.from_name(REPORTER_ENDPOINT))
+
+            reporter = event_bus._outbound_connections[REPORTER_ENDPOINT]
+            await reporter.wait_until_subscribed_to(TotalRecordedEvent)
+
+            counter = itertools.count(1)
+            stats = LocalStatistic()
+            async for event in event_bus.stream(PerfMeasureEvent):
```

Comment:
Is there a reason this doesn't this use the `num_events` API that the `stream` method implements (instead of the current thing which uses an `itertools.count()` to exit the loop?  Then you could move the broadcasting of the `TotalRecordedEvents` into an `else` clause (or just after this for loop's block)

Reply:
Good catch! Did this in a different PR